pkgs/by-name/go/golden-cheetah-bin/darwin.nix

```
@@ -0,0 +1,35 @@
+{
+  fetchurl,
+  stdenv,
+  makeWrapper,
+  undmg,
+  pname,
+  meta,
+  version,
+}:
+
+stdenv.mkDerivation {
+  inherit
+    pname
+    version
+    meta
+    ;
+
+  src = fetchurl {
+    url = "https://github.com/GoldenCheetah/GoldenCheetah/releases/download/v${version}/GoldenCheetah_v${builtins.substring 0 7 version}_x64.dmg";
+    hash = "sha256-GGe8pGf16FWqN9uaStnGOtOeKDB44VBD7QwHkaZb7wY=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    undmg
+  ];
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+    mkdir -p "$out/Applications/GoldenCheetah.app"
+    cp -r * "$out/Applications/GoldenCheetah.app/"
+    makeWrapper "$out/Applications/GoldenCheetah.app/Contents/MacOS/GoldenCheetah" "$out/bin/GoldenCheetah"
+    runHook postInstall
+  '';
```

```suggestion
  '';
  
  meta = {
    inherit
      (meta)
        description
        platforms
        maintainers
        license
        sourceProvenance
        ;
  };
```

While being very, very annoying, this helps make sure things like `meta.pos` are still set correctly, allowing for commands like `nix edit` to work

---

ah, ok. Done

---

Seems like this causes an
>        error: infinite recursion encountered

---

Ah that's due to the use of `rec` in the other file. Could you change the name of the `meta` argument passed to `callPackage` to something else instead -- maybe `commonMeta`? This would avoid the naming conflict

---

Ah, thanks for the pointer. I was rather clueless. Changed and works 👍